docs/source/libmav/libmav.py

```diff
@@ -0,0 +1,1041 @@
+
+        """
+        pass
+
+        
+    def keys(self):
+        """Returns a list of the payload keys in the message (the same information as `field_names()`).
+        
+        Returns:
+            list: A list of all the field names in the message.
+
+        """
+        pass
+
+
+class MessageSet():
+    """A class representing a set of MAVLink message and enum definitions.
+    
+    The object can be be populated from a MAVLink XML definition file, in which case it will automatically pull in any included XML definitions.
+    It can also be populated from a string containing an XML definition.
+    
+    `MessageSet` can be used to create `Message` instances for sending, such as the `HEARTBEAT`.
+    It is also passed to the `NetworkRuntime` on construction, providing the definitions that libmav uses in order to parse MAVLink messages.
+    
+    ```python
+    # Construct the common message set
+    message_set = libmav.MessageSet('./mavlink/message_definitions/v1.0/common.xml')
+
+    # Check the id and create a message from it
+    message_id = heartbeat_message.id_for_message('HEARTBEAT')
+    heartbeat_message = message_set.create(message_id)
+        
+    # Create a HEARTBEAT message from its name.       
+    heartbeat_message = message_set.create('HEARTBEAT')  
+    ```
+    
+    Note that the new messages are created with default values.
+    
+    """  
+
+    def __init__(self):
+        """Construct an empty `MessageSet`.
+        
+        ```python
+        # Construct an empty message set
+        message_set = libmav.MessageSet()
+        ```
+        
+        Definitions can be added to the new set using `add_from_xml_file()` or `add_from_xml_string()`.
+
+        Returns:
+            MessageSet: An empty `MessageSet`.
+        """
+        pass
+        
+    def __init__(self, definition_file):
+        """Construct an empty `MessageSet`.
+        
+        Additional definitions can be added to the new set using `add_from_xml_file()` or `add_from_xml_string()`.
+        <!-- what happens if there are clashes/overloads?? -->
+
+        ```python
+        # Construct the common message set
+        message_set = libmav.MessageSet('./mavlink/message_definitions/v1.0/common.xml')
+        
+        # Construct the ardupilotmega message set. 
+        message_set = libmav.MessageSet('./mavlink/message_definitions/v1.0/ardupilotmega.xml')
+        ```
+
+        Args:
+            definition_file (str): Full path to the XML definition file to load.
+
+        Returns:
+            MessageSet: A new `MessageSet` object populated with the definitions in the indicated file.
+        """
+        pass        
+             
+
+    def add_from_xml_file(self, definition_file):
+        """Add definitions from a MAVLink XML definition file.
+        
+        The new definitions are appended to the  to the `MessageSet`.
+        Overloaded definitions are replaced. <!-- is this correct? --> 
```

Comment:
Is this correct? I.e. additions are appended and overloads replace the original definitions?

Reply:
Correct